lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -156,11 +155,15 @@ def do_grant(kms, keyarn, role_arn, granttypes, mode='grant', dry_run=True, clea
             # do we want this grant type? Are we on its statement?
             # and does the role have this grant type?
 
-            # Ensure statement looks as expected
-            if not statement.get('Principal'):
-                statement['Principal'] = {'AWS': []}
-            if not isinstance(statement['Principal']['AWS'], list):
-                statement['Principal']['AWS'] = [statement['Principal']['AWS']]
+            # create Principal and 'AWS' so we can safely use them later.
+            if not isinstance(statement.get('Principal'), dict):
+                statement['Principal'] = dict()
+
+            if 'AWS' in statement['Principal'] and isinstance(statement['Principal']['AWS'], str):
+                # convert to list
+                statement['Principal']['AWS'] = list([statement['Principal']['AWS']])
```

Comment:
This is creating a list `[statement['Principal']['AWS']]` and then passing that to `list()`. Can you either remove the surrounding `[]` or the `list()` call?

Reply:
fixed, thanks.